### ISSUE TITLE
OpenVINO EP is not portable

### ISSUE THREAD

**Issue by GuanLuo** [microsoft/onnxruntime#1861]
**Is your feature request related to a problem? Please describe.**
I built ONNX Runtime with OpenVINO, but the OpenVINO EP uses [PyImport_ImportModule](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/openvino/openvino_graph.cc#L143) on the [openvino_mo](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/openvino/openvino_mo/openvino_mo.py) script. The script is copied to the same build directory as the ONNX Runtime library, which is easy to find, however, the import module function will only look up the module in current directory, if not present, OpenVINO will give the following error:
```
2019-09-17 17:28:47.736465650 [W:onnxruntime:test, openvino_execution_provider.cc:491 GetCapability] [OpenVINO-EP] Rejecting as Model Optimizer cannot convert this model.Python module import failure
```
In other words, the user will have to make sure the application is run with the scripts in the current directory. Please let me know if I missed something which can relieve me from doing so.

**System information**
- ONNX Runtime version (you are using): master

**Describe the solution you'd like**
An API to specify where the import function should look up the scripts.

**Comment by jywu-msft** [comment]
@smkarlap , @suryasidd

**Comment by smkarlap** [comment]
Yes, it seems like the python interpreter doesn't know the location of `openvino_mo.py` and `openvino_emitter.py` scripts.

`openvino_mo.py` and `openvino_emitter.py` are the python scripts required for using the OpenVINO EP, and are invoked by the python interpreter from within OpenVINO EP.  So, the location of these two files must be known to the python interpreter. This can be achieved in two ways:-

**1.  Installing the .whl package**
   When you build ONNX Runtime with the `--build-wheel` option, a .whl package is generated in the `<ONNX_RUNTIME_ROOT>/build/<OS>/<build_type>/dist` directory. This .whl package contains all the libraries and scripts required for running the application, and can be installed using `pip install` command which would install the libraries and scripts to known locations. As long as the python environment of `pip` and runtime python interpreter is the same, the runtime python interpreter will know where to find the files installed by `pip`. So in this case, nothing else needs to be done for the `openvion_mo.py` and `openvino_emitter.py` scripts to be accessible from anywhere.

**2. Directly using the executables from the build folder**
  If you have not installed the `.whl` package and would like to run the application by accessing the executables and libraries in the `<ONNX_RUNTIME_ROOT>/build` folder, then the location of the python scripts needs to be added to the PYTHONPATH environment variable. Python interpreter searches all the directories listed in this variable for modules/scripts. This can be achieved with the command:-
   `export PYTHONPATH=<full_directory_path_to_the_py_scripts>:$PYTHONPATH`

**Comment by GuanLuo** [comment]
@smkarlap Thanks, setting `PYTHONPATH` solves my problem, closing the issue. I am also wondering if there is plan to separate the Python dependency from the library, I was using the ONNX Runtime C library in a docker image that doesn't have Python, and now I have to install Python utilities to use the OpenVINO EP...

**Comment by smkarlap** [comment]
Python dependency will be needed, at least in the near future, to infer ONNX models on the OpenVINO stack. We will let you know once we're able to remove this dependency.
